libr/bin/format/pe/pe.c

```diff
@@ -2264,6 +2265,16 @@ static void _parse_resource_directory(struct PE_(r_bin_pe_obj_t) *bin, Pe_image_
 			eprintf ("Warning: read resource entry\n");
 			break;
 		}
+		ut8 *resourceEntryName = NULL;
+		if (entry.u1.s.NameIsString) {
+			ut16 resourceEntryNameLength;
+			r_buf_read_at (bin->b, bin->resource_directory_offset + entry.u1.s.NameOffset, (ut8*)&resourceEntryNameLength, sizeof (ut16));
+
+			resourceEntryName = calloc (resourceEntryNameLength, sizeof (ut8));
```

Comment:
leak?

Reply:
For "normal" PEs, the resourceEntryName will be passed to `_parse_resource_directory()` again in the last parameter and assigned to "name" in the `r_pe_resource` struct which will be freed in `_free_resources()`. I have to do it this way since the name is actually the name of the directory and not the name of the resource itself (see last picture of @ITAYC0HEN in his issue).

But I agree that there might be a memory leak if you have weird resource structures. I'll add a free in the cases where the resourceEntryName wouldn't end up in `r_pe_resource`.